doc/manual.rst

```diff
@@ -7894,6 +7894,9 @@ instructs the compiler to pass the type by value to procs:
     Vector {.bycopy.} = object
       x, y, z: float
 
+Nim will pass objects larger than 24 bytes by reference for performance. 
+Using `bycopy` disables this. The `bycopy` can be used with `exportc` 
+when interfacing with C to match the function signature exactly. 
```

Comment:
1. It's not 24 bytes, 3*sizeof(int) and it's implementation defined.
2. This should be added to the `exportc` section.
3. Even with `bycopy` plus `exportc` it's not good enough, you also need the `cdecl` calling convention. This should be added to the `exportc` section. 

However, it can be quite exhausting to write down what pragmas *don't* do as it implies an infinite amount of documentation work...

Reply:
1. I don't think so, its `(getSize(conf, pt) > conf.target.floatSize * 3)` and floatSize is float 64 on all platforms https://github.com/nim-lang/Nim/blob/b3aca78e228feb93d5dcd6f00cdd81e4178c24cc/compiler/platform.nim#L205 . I think its effectively 3*sizeof(float64) which is pretty constant unlike 32/64 bit ints would. Even on 16 bit platforms c's double was always 64 bytes. I could say that its "3*sizeof(floatSize) and floatSize is 64 bits making it 24 bytes on all current supported platforms but its subject to change if new platform is added in the future" 
2/3. You want some thing like "to export X properly to do these things: exportc bycopy cdecl"?

Reply:
Regardless, I don't think it's a good idea to specifically state at what size a parameter is passed by value vs by reference. 
Something like:
```
Note: The Nim compiler automatically determines whether a parameter is passed by value or by reference based on factors such as the parameter type's size. If a parameter *must* be passed by value or by reference, use the `bycopy` or `byref` pragmas.
```